Test/Tests/RealtimeClientPresenceTests.swift

```diff
@@ -756,86 +756,106 @@ class RealtimeClientPresenceTests: XCTestCase {
         }
     }
 
-    func test__023__Presence__Channel_state_change_side_effects__channel_enters_the_SUSPENDED_state__should_maintain_the_PresenceMap_and_any_members_present_before_and_after_the_sync_should_not_emit_presence_events() throws {
+    func test__023__Presence__Channel_state_change_side_effects__channel_enters_the_SUSPENDED_state__members_map_is_preserved_and_only_members_that_changed_between_ATTACHED_states_should_result_in_presence_events() throws {
         let test = Test()
         let options = try AblyTests.commonAppSetup(for: test)
         let channelName = test.uniqueChannelName()
 
-        var clientMembers: ARTRealtime?
-        clientMembers = AblyTests.addMembersSequentiallyToChannel(channelName, members: 3, options: options)
-        defer { clientMembers?.dispose(); clientMembers?.close() }
-
-        options.clientId = "tester"
+        let clientMembers = AblyTests.addMembersSequentiallyToChannel(channelName, members: 2, options: options)
+        defer { clientMembers.dispose(); clientMembers.close() }
+        
+        options.clientId = "leaves"
+        let leavesClient = AblyTests.newRealtime(options).client
+        defer { leavesClient.dispose(); leavesClient.close() }
+        
+        options.clientId = "main"
         options.tokenDetails = try getTestTokenDetails(for: test, key: options.key!, clientId: options.clientId, ttl: 5.0)
-        let client = AblyTests.newRealtime(options).client
-        defer { client.dispose(); client.close() }
-        let channel = client.channels.get(channelName)
+        let mainClient = AblyTests.newRealtime(options).client
+        defer { mainClient.dispose(); mainClient.close() }
+        
+        let leavesChannel = leavesClient.channels.get(channelName)
+        let mainChannel = mainClient.channels.get(channelName)
+        
         waitUntil(timeout: testTimeout) { done in
-            let partialDone = AblyTests.splitDone(3, done: done)
-            channel.presence.enter(nil) { error in
+            let partialDone = AblyTests.splitDone(5, done: done)
+            mainChannel.presence.enter(nil) { error in
                 XCTAssertNil(error)
                 partialDone()
             }
-            channel.presence.get { members, error in
+            leavesChannel.presence.enter(nil) { error in
+                XCTAssertNil(error)
+                partialDone()
+            }
+            mainChannel.presence.get { members, error in
                 XCTAssertNil(error)
                 if let members {
-                    XCTAssertEqual(members.count, 3)
+                    XCTAssertEqual(members.count, 3) // "user1", "user2", "leaves"
```

Comment:
Is this guaranteed to contain `"leaves"`, given that line 781's `leavesChannel.presence.enter` may not yet even have completed?

Reply:
https://github.com/ably/ably-cocoa/pull/1793/commits/5220e18b399095293657b1e269566226d4c9ce5f